src/Compilers/CSharp/Portable/BoundTree/BoundNode.cs

```diff
@@ -149,6 +150,25 @@ public void ResetCompilerGenerated(bool newCompilerGenerated)
             }
         }
 
+        public bool IsSuppressed
+        {
+            get
+            {
+                return (_attributes & BoundNodeAttributes.IsSuppressed) != 0;
+            }
+            internal set
+            {
```

Comment:
The existence of this internal setter is inherently dangerous.  Can you make it protected?  Or better yet remove it and instead add a protected method whose purpose is to set suppressed to true.